vita3k/host/src/pup.cpp

```diff
@@ -253,25 +253,34 @@ void install_pup(const std::string &pup, const std::string &pref_path) {
 
     if (fs::file_size(pup_dec + "/os0.img") > 0) {
         extract_fat(pup_dec, "os0.img", pref_path);
+        for (const auto &file : fs::recursive_directory_iterator(pref_path + "/os0")) {
+            if (fs::is_regular_file(file.path())) {
+                const auto extention = file.path().filename().extension();
+                const auto is_self = ((extention == ".suprx") || (extention == ".skprx") || (extention == ".self"));
+                if (is_self) {
+                    self2elf(file.path().string(), file.path().string() + "elf", SCE_KEYS, 0);
+                    fs::rename(file.path().string() + "elf", file.path().string());
+                    make_fself(file.path().string(), file.path().string() + "fself");
+                    fs::rename(file.path().string() + "fself", file.path().string());
+                }
+            }
+        }
     }
-    if (fs::file_size(pup_dec + "/sa0.img") > 0) {
+    if (fs::file_size(pup_dec + "/sa0.img") > 0)
         extract_fat(pup_dec, "sa0.img", pref_path);
-    }
     if (fs::file_size(pup_dec + "/vs0.img") > 0) {
         extract_fat(pup_dec, "vs0.img", pref_path);
-        for (const auto &file : fs::recursive_directory_iterator(pref_path + "/vs0/app")) {
-            if (file.path().filename() == "eboot.bin") {
-                self2elf(file.path().string(), file.path().string() + "elf", SCE_KEYS, 0);
-                fs::rename(file.path().string() + "elf", file.path().string());
-                make_fself(file.path().string(), file.path().string() + "fself");
-                fs::rename(file.path().string() + "fself", file.path().string());
+        for (const auto &file : fs::recursive_directory_iterator(pref_path + "/vs0")) {
+            if (fs::is_regular_file(file.path())) {
+                const auto extention = file.path().filename().extension();
+                const auto is_self = ((extention == ".suprx") || (extention == ".skprx") || (extention == ".self"));
+                if ((file.path().filename() == "eboot.bin") || is_self) {
+                    self2elf(file.path().string(), file.path().string() + "elf", SCE_KEYS, 0);
+                    fs::rename(file.path().string() + "elf", file.path().string());
+                    make_fself(file.path().string(), file.path().string() + "fself");
+                    fs::rename(file.path().string() + "fself", file.path().string());
+                }
```

Comment:
Extract `file.path().string()` lol, you use it like 17 times here and lines of code above.

I suggest something like the following with better naming lol
```
        for (const auto &file : fs::recursive_directory_iterator(pref_path + "/vs0")) {
            fs::path filePath = file.path();
            std::string fileString = filePath.string();

            if (fs::is_regular_file(filePath) {
                const auto extention = filePath.filename().extension();
                const auto is_self = ((extention == ".suprx") || (extention == ".skprx") || (extention == ".self"));
                if ((file.path().filename() == "eboot.bin") || is_self) {
                    self2elf(fileString, file.path().string() + "elf", SCE_KEYS, 0);
                    fs::rename(fileString + "elf", fileString);
                    make_fself(fileString, fileString + "fself");
                    fs::rename(fileString + "fself", fileString);
                }
            }
        }
```

Reply:
hmm i don't like lokl is more ugly with more create variable not needed ^^

Reply:
code for this is untouched, i have just added for check extension file :)

Reply:
i have change one things like you suggest :)